### ISSUE TITLE
Building on top of the Wayland Client Window Example

### ISSUE THREAD

**Issue by chrisduerr** [vberger/wayland-client-rs#76]
I am trying to create an app using this Wayland crate in Rust and ran into the issue that I find the example very hard to build up on.

My goal is to implement the `geometry` method of `wl_output::Handler` so I would be able to get the current dimensions of the window. Implementing the handler itself is pretty straight forward, but registering it on the `EventQueue` seems a little more difficult.

In the example everything is first created in a separate scope and after that the scope is closed and the objects are added. This works since the methods like `env.seat.get_pointer()` return new objects and don't rely on the `env` anymore. If I want to use `env.output` however I always need to keep the `env` around which will lead to issues since during the creation of `env` `event_queue` is mutably borrowed and `event_queue.register` requires a mutable borrow, too.

My current code attempt looks like this: <http://hastebin.com/uzojuxukuc.rs>
All the other stuff like `wayland_env!` and `declare_handler!` should be fine.

Looking at the example getting the width and height of an output seems really awkward. Is there a simple way to solve this issue?

**Comment by vberger** [comment]
I agree I need to make more examples, this is on my todo-list.

Regarding your specific problem with outputs and ownership, be aware that you can re-instanciate globals as often as needed, using the `WlRegistry::bind(..)` method, using the values provided by the `Envhandler::globals(..)` method. Is is perfectly safe and find to duplicate globals this ways. The wayland_env!() macro is just a wrapper doing part of this job for you, for the globals you need only once at initialization.

**Comment by vberger** [comment]
I don't have a lot of time right now (until next week). You can have a look at wayland-window's code for a less basic example of how the lib can be used.

**Comment by chrisduerr** [comment]
Thanks for the quick reply. I was able to duplicate the object by using the result of `WlRegistry::bind(..)`. If you have the time for a few more examples I'm sure a lot of people would appreciate that!

**Comment by vberger** [comment]
I've created #78 as a follow-up issue about writing more examples. If you have some ideas about what examples would be helpful, please leave a note there about it! :)
